app/components/project-category-item.js

```diff
@@ -1,13 +1,59 @@
 import Ember from 'ember';
 
+/**
+  `project-category-item` represents a category with a project. When selected
+  the user is taken to a list of projects using this category.
+
+  ## Default usage
+  ```Handlebars
+  {{project-category-item category=category}}
+  ```
+
+  @class project-category-item
+  @module Components
+  @extends Ember.Component
+ */
 export default Ember.Component.extend({
   tagName: ['li'],
 
+  /**
+    Returns true if 'userCategory' is not empty
+
+    @property selected
+    @type Boolean
+   */
   selected: Ember.computed.notEmpty('userCategory'),
+
+  /**
+    A service that returns all userCategories.
+
+    @property userCategories
+    @type Ember.Service
+   */
   userCategories: Ember.inject.service(),
+
+  /**
+    The current user.
+
+    @property user
+    @type Ember.Object
```

Comment:
I would vote for `DS.Model` here, in favor of specificity. If you're diving into docs and want to know how something works `Ember.Object` isn't particularly revealing. I'd like to be able to jump into `ember-data` instead and see what methods are available on this object.